modules/graph/vertex_map/arrow_vertex_map.h

```diff
@@ -308,6 +412,70 @@ class ArrowVertexMap<arrow::util::string_view, VID_T>
     return static_cast<vid_t>(oid_arrays_[fid][label_id]->length());
   }
 
+  ObjectID AddVertices(
+      Client& client,
+      const std::vector<std::vector<std::shared_ptr<oid_array_t>>>&
+          oid_arrays) {
+    size_t extra_label_num = oid_arrays.size();
+
+    std::vector<std::vector<typename InternalType<oid_t>::vineyard_array_type>>
+        vy_oid_arrays;
+    int total_label_num = label_num_ + extra_label_num;
+    vy_oid_arrays.resize(fnum_);
+    for (fid_t i = 0; i < fnum_; ++i) {
+      vy_oid_arrays[i].resize(extra_label_num);
+    }
+
+    ThreadGroup tg;
+    auto builder_fn = [this, &client, &oid_arrays, &vy_oid_arrays](
+                          fid_t const fid,
+                          label_id_t const vlabel_id) -> Status {
+      auto& array = oid_arrays[vlabel_id][fid];
+      typename InternalType<oid_t>::vineyard_builder_type array_builder(client,
+                                                                        array);
+      vy_oid_arrays[fid][vlabel_id] = *std::dynamic_pointer_cast<
+          typename InternalType<oid_t>::vineyard_array_type>(
+          array_builder.Seal(client));
+      return Status::OK();
+    };
+
+    for (fid_t fid = 0; fid < fnum_; ++fid) {
+      for (label_id_t vlabel_id = 0; vlabel_id < extra_label_num; ++vlabel_id) {
+        tg.AddTask(builder_fn, fid, vlabel_id);
+      }
+    }
```

Comment:
Do we need to build the whole vertexmap, rather than just for the newly added vertices?